File: 2018_Quals/B/Solution.java
import java.util.*;
import java.io.*;
import java.math.*;
import java.awt.geom.*;
public class Solution {
  public static final BufferedReader buffer = new BufferedReader(new InputStreamReader(System.in));
  public static final PrintWriter outWriter = new PrintWriter(System.out);
  public static Object solve() {
    int[] in = new int[nextInt()];
    for (int i = 0; i < in.length; i++) { in[i] = nextInt(); }
    boolean flag;
    do {
      flag = false;
      for (int i = 0; i < in.length-2; i++) {
        if (in[i] > in[i+2]) {
          in[i]^=in[i+2]^(in[i+2]=in[i]);
          flag = true;
        }
      }
    } while (flag);
    for (int i = 0; i < in.length-1; i++) {
      if (in[i] > in[i+1]) {
        return i;
      }
    }
    return "OK";
  }
  public static void main(String[] args) {
    int T = nextInt();
    for (int i = 0; i < T; i++) {
      outWriter.print("Case #"+(i+1)+": ");
      Object tmp = solve();
      if (tmp != null) { outWriter.println(tmp); }
    }
    outWriter.flush();
  }
  public static StringTokenizer tokenizer = null;
  public static String nextLine() {
    try { return buffer.readLine(); } catch (IOException e) { throw new UncheckedIOException(e); }
  }
  public static String next() {
    while (tokenizer == null || !tokenizer.hasMoreElements()) { tokenizer = new StringTokenizer(nextLine()); }
    return tokenizer.nextToken();
  }
  public static int nextInt() { return Integer.parseInt(next()); }
  public static long nextLong() { return Long.parseLong(next()); }
  public static double nextDouble() { return Double.parseDouble(next()); }
}

File: 2018_Quals/B/Solution.c
#include <stdio.h>
#define MAX 100000

void solve() {
  int n;
  int in[MAX];
  scanf("%d", &n);
  
  for (int i = 0; i < n; i++) {
    scanf("%d", in+i);
  }
  int flag, tmp;
  do {
    flag = 0;
    for (int i = 0; i < n-2; i++) {
      if (in[i] > in[i+2]) {
        tmp = in[i];
        in[i] = in[i+2];
        in[i+2] = tmp;
        flag = 1;
      }
    }
  } while (flag);
  for (int i = 0; i < n-1; i++) {
    if (in[i] > in[i+1]) {
      printf("%d\n", i);
      return;
    }
  }
  puts("OK");
}

int main() {
  int t;
  scanf("%d", &t);
  for (int i = 1; i <= t; i++) {
    printf("Case #%d: ", i);
    solve();
  }
}
